src/sql/base/browser/ui/table/plugins/headerFilter.plugin.ts

```diff
@@ -32,12 +32,18 @@ export interface CommandEventArgs<T extends Slick.SlickData> {
 export interface ITableFilterStyles extends IButtonStyles, IInputBoxStyles, IListStyles, ICountBadgetyles {
 }
 
+interface NotificationProvider {
```

Comment:
What's the purpose of defining a separate interface for this instead of just having the constructor take in a service directly? Do you foresee having to support other types of notifications?

Reply:
this is in the base layer, the notification service definition is not available here, for now, I think a simple notification is good enough, in the future, if more complicated handler is required, i can expose an event for the consumer of this plugin to use.